src/main/java/com/sap/cloudfoundry/client/facade/adapters/CloudFoundryClientFactory.java

```diff
@@ -54,8 +58,17 @@ private ConnectionContext createConnectionContext(String controllerApiHost) {
         getConnectTimeout().ifPresent(builder::connectTimeout);
         getConnectionPoolSize().ifPresent(builder::connectionPoolSize);
         getThreadPoolSize().ifPresent(builder::threadPoolSize);
-        builder.additionalHttpClientConfiguration(client -> client.metrics(true));
+        builder.additionalHttpClientConfiguration(this::getAdditionalHttpClientConfiguration);
         return builder.build();
     }
 
+    private HttpClient getAdditionalHttpClientConfiguration(HttpClient client) {
+        HttpClient clientWithOptions = client;
```

Comment:
do you need this?

Reply:
I need this, because everytime when you set some configuration of HttpClient, new object instance is created and I have to assign it to variable.